extensions/resteasy-reactive/quarkus-resteasy-reactive-kotlin-serialization/runtime/src/main/kotlin/io/quarkus/kotlin/serialization/KotlinSerializationMessageBodyReader.kt

```diff
@@ -0,0 +1,44 @@
+package io.quarkus.kotlin.serialization
+
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.serializer
+import org.jboss.resteasy.reactive.common.providers.serialisers.AbstractJsonMessageBodyReader
+import org.jboss.resteasy.reactive.common.util.StreamUtil
+import org.jboss.resteasy.reactive.server.spi.ResteasyReactiveResourceInfo
+import org.jboss.resteasy.reactive.server.spi.ServerMessageBodyReader
+import org.jboss.resteasy.reactive.server.spi.ServerRequestContext
+import java.io.InputStream
+import java.lang.reflect.Type
+import javax.inject.Inject
+import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.MultivaluedMap
+
+class KotlinSerializationMessageBodyReader(@Inject var json: Json) : AbstractJsonMessageBodyReader(), ServerMessageBodyReader<Any> {
```

Comment:
If you look at `BasicServerJacksonMessageBodyWriter`  it implements `AllWriteableMessageBodyWriter` which allows for an optimization during MessageBodyWriter lookup. I think we can do the same here

Reply:
Scratch that, my bad